imsim/photon_ops.py

```diff
@@ -275,7 +241,9 @@ def config_kwargs(config, base, cls):
     """
     req, opt, single, _takes_rng = get_cls_params(cls)
     kwargs, _safe = GetAllParams(config, base, req, opt, single)
-    opt_kwargs = specified_opt_kwargs(kwargs, cls._opt_params.keys())
+    opt_kwargs = {
+        key: kwargs.get(key) for key in cls._opt_params.keys() if key in kwargs
+    }
     return kwargs, opt_kwargs
```

Comment:
I think this would be cleaner to only return `kwargs`, not separate out `opt_kwargs` explicitly.
Then any kwargs that need special handling, you can pop off (e.g. `telescope` and `band`).  The ones that are used as is (e.g. boresight and any optional ones) get naturally applied by ending the call to `LsstOptics` with `**kwargs`.

Reply:
Done